imap module - local maildir implementation

mrt-prodz opened the pull request:
Following this issue: https://github.com/ultrabug/py3status/issues/143

This is a bit messy but it would add new emails to the unseen mail count if a user is having both offline and remote mailbox. If the user only wants to check for offline maildir he can set the offline parameter to '1'.

User can also specifiy multiple local maildir using comma separated full path. Maybe semi-colon would be more appropriate. It seems to work properly here but I would really like your input on this since it doesn't look really clean (at all).

Should the failure of retrieving local maildir stop the module from trying to retrive remote emails as well?

tobes commented on py3status/modules/imap.py:
```diff
@@ -9,7 +9,9 @@
     mailbox: name of the mailbox to check
+    maildir: full path of your maildir location
     new_mail_color: what color to output on new mail
+    offline: set to 1 will only check for local maildir
```
Maybe instead of this option just see if we have imap_server set, if imap_server == '<IMAP_SERVER>' then nothing remote to check.

Less config options is always nice.

mrt-prodz commented:
Removed the `offline` variable and checking instead if `maildir` or `imap_server` have been declared.
I also replaced the comma separated maildir list with a semi-colon.

mrt-prodz commented:
local mbox implementation, similar to maildir we can add multiple paths using semi-colon as separator.

tobes commented:
> I also replaced the comma separated maildir list with a semi-colon.

Having `,` used in one places (mailbox) and `;` in others seems confusing to me.  Since `,` was there already I think sticking with that might be better

pydsigner commented:
@tobes The problem with commas is that they are valid characters in a filename. Colons make the most sense IMO.

mrt-prodz commented:
Colons would be the more appropriate of all regarding multiple paths, like how the environment variable $PATH is set for example.

tobes commented:
> Colons would be the more appropriate of all regarding multiple paths, like how the environment variable $PATH is set for example.

In master we can now use lists directly in the config eg `mailbox = ['box 1', 'box 2']`

this module should probably be adapted to use them - but maybe this PR has stalled

raspbeguy commented:
@mrt-prodz I would love to see this PR being merged. This is a good idea! Please continue to work on it.

mrt-prodz commented:
@raspbeguy I'll try to find some time this weekend to update this PR, I guess it needs some refactoring to meet recent module implementation. It seemed to work properly but I really need people to test it.

tobes commented:
We should either close this or get it cleaned up and merged

lasers commented:
@mrt-prodz, Hi! 😊 Can you give us an update on things here? Do you still use this module? Maybe this could be its own module since it does not have anything to do with `imap`? And that we got another (open) `imap` pull request too?

Knowing you use `thunderbird`, do you use this with `thunderbird` too? Maybe I can try and update this module too.

mrt-prodz commented:
@lasers , it was mostly an attempt at solving issue #143 . I don't use maildir anymore that's why I wanted someone with real data to test it. It seems there is not much interest with local maildir implementation though so I'm not sure if it's something useful/needed.
